### PR TITLE
main, service: allow and default to Docker Hub

### PR DESCRIPTION
**Pull request by kopiczko**
## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/internal/cloudconfig/template/aws-cni.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ spec:
       tolerations:
         - operator: Exists
       containers:
-        - image: {{.RegistryDomain}}/giantswarm/amazon-k8s-cni:v1.6.0
+        - image: {{ if ne .RegistryDomain "" }}{{ .RegistryDomain }}/{{ end }}/giantswarm/amazon-k8s-cni:v1.6.0
```
Comment (corest) [inline service/internal/cloudconfig/template/aws-cni.go:104]:
```suggestion
        - image: {{ if ne .RegistryDomain "" }}{{ .RegistryDomain }}/{{ end }}giantswarm/amazon-k8s-cni:v1.6.0
```

File: service/internal/cloudconfig/template/aws-cni.go [hunk 2]
```diff
@@ -295,7 +295,7 @@ spec:
           hostNetwork: true
           containers:
             - name: aws-cni-restarter
-              image: {{.RegistryDomain}}/giantswarm/aws-cni-restarter:1.0.2
+              image: {{ if ne .RegistryDomain "" }}{{ .RegistryDomain }}/{{ end }}/giantswarm/aws-cni-restarter:1.0.2
```
Comment (corest) [inline service/internal/cloudconfig/template/aws-cni.go:298]:
```suggestion
              image: {{ if ne .RegistryDomain "" }}{{ .RegistryDomain }}/{{ end }}giantswarm/aws-cni-restarter:1.0.2
```

File: service/internal/cloudconfig/template/decrypt_keys_assets_script.go [hunk 3]
```diff
@@ -2,7 +2,7 @@ package template
 
 const DecryptKeysAssetsScript = `#!/bin/bash -e
 kms_keys_assets_decrypt() {
-AWS_CLI_IMAGE="{{.RegistryDomain}}/giantswarm/awscli:1.18.3"
+AWS_CLI_IMAGE="{{ if ne .RegistryDomain "" }}{{ .RegistryDomain }}/{{ end }}/giantswarm/awscli:1.18.3"
```
Comment (corest) [inline service/internal/cloudconfig/template/decrypt_keys_assets_script.go:5]:
```suggestion
AWS_CLI_IMAGE="{{ if ne .RegistryDomain "" }}{{ .RegistryDomain }}/{{ end }}giantswarm/awscli:1.18.3"
```

File: service/internal/cloudconfig/template/decrypt_tls_assets_script.go [hunk 4]
```diff
@@ -2,7 +2,7 @@ package template
 
 const DecryptTLSAssetsScript = `#!/bin/bash -e
 kms_tls_assets_decrypt() {
-AWS_CLI_IMAGE="{{.RegistryDomain}}/giantswarm/awscli:1.18.3"
+AWS_CLI_IMAGE="{{ if ne .RegistryDomain "" }}{{ .RegistryDomain }}/{{ end }}/giantswarm/awscli:1.18.3"
```
Comment (corest) [inline service/internal/cloudconfig/template/decrypt_tls_assets_script.go:5]:
```suggestion
AWS_CLI_IMAGE="{{ if ne .RegistryDomain "" }}{{ .RegistryDomain }}/{{ end }}giantswarm/awscli:1.18.3"
```

File: service/internal/cloudconfig/template/etcd3_attach_dep_service.go [hunk 5]
```diff
@@ -9,7 +9,7 @@ Before=k8s-kubelet.service
 
 [Service]
 # image is from https://github.com/giantswarm/aws-attach-etcd-dep
-Environment="IMAGE={{ .RegistryDomain }}/giantswarm/aws-attach-etcd-dep:27182d5b07d824576dd47869e5f921c1341fb194"
+Environment="IMAGE={{ if ne .RegistryDomain "" }}{{ .RegistryDomain }}/{{ end }}/giantswarm/aws-attach-etcd-dep:27182d5b07d824576dd47869e5f921c1341fb194"
```
Comment (corest) [inline service/internal/cloudconfig/template/etcd3_attach_dep_service.go:12]:
```suggestion
Environment="IMAGE={{ if ne .RegistryDomain "" }}{{ .RegistryDomain }}/{{ end }}giantswarm/aws-attach-etcd-dep:27182d5b07d824576dd47869e5f921c1341fb194"
```

File: go.mod [hunk 6]
```diff
@@ -20,7 +20,7 @@ require (
 	github.com/giantswarm/helmclient v1.0.3
 	github.com/giantswarm/ipam v0.2.0
 	github.com/giantswarm/k8sclient/v3 v3.1.0
-	github.com/giantswarm/k8scloudconfig/v6 v6.2.6
+	github.com/giantswarm/k8scloudconfig/v6 v6.2.7-0.20200629152348-2147bb11c876
```
Comment (xh3b4sd) [inline go.mod:23]: TODO

File: main.go [hunk 7]
```diff
@@ -154,7 +154,7 @@ func mainE(ctx context.Context) error {
 	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.TLS.CrtFile, "", "Certificate file path to use to authenticate with Kubernetes.")
 	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.TLS.KeyFile, "", "Key file path to use to authenticate with Kubernetes.")
 
-	daemonCommand.PersistentFlags().String(f.Service.RegistryDomain, "quay.io", "Image registry.")
+	daemonCommand.PersistentFlags().String(f.Service.RegistryDomain, "", "Image registry. Empty value will cause using Docker Hub.")
```
Comment (xh3b4sd) [inline main.go:157]: I wouldn't add this assumption here. The default may change and nobody will fix this here.

Comment (kopiczko) [reply main.go:157]: We use the configuration from ConfigMap so I think we are good here. I just wanted to remove quay reference as we are going to stop using it.

**Comment by kopiczko** [comment]
We don't need that. Mirroring works with `docker.io`.

**Pull request closed without merging**